components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+                    </div>
+                    <div className="pt-3">
+                        {errorIde && <div className="text-red-500 text-sm">{errorIde}</div>}
+                        <SelectIDEComponent
+                            onSelectionChange={onSelectEditorChange}
+                            setError={setErrorIde}
+                            selectedIdeOption={selectedIde}
+                            useLatest={useLatestIde}
+                        />
+                    </div>
+                    <div className="pt-3">
+                        {errorWsClass && <div className="text-red-500 text-sm">{errorWsClass}</div>}
+                        <SelectWorkspaceClassComponent
+                            onSelectionChange={setSelectedWsClass}
+                            setError={setErrorWsClass}
+                            selectedWorkspaceClass={selectedWsClass}
+                        />
+                    </div>
+                </div>
+                <div className="w-full flex justify-end mt-6 space-x-2 px-6">
+                    <button
+                        key="start"
+                        className=""
+                        onClick={onClickCreate}
+                        disabled={creating || !repo || repo.length === 0 || !!errorIde || !!errorWsClass}
+                    >
+                        {creating ? (
+                            <div className="flex">
+                                <img className="h-4 w-4 animate-spin" src={Spinner} alt="loading spinner" />
+                                <span className="pl-2">Creating Workspace ...</span>
+                            </div>
+                        ) : (
+                            "New Workspace"
+                        )}
+                    </button>
+                </div>
+                <div>
+                    <StatusMessage
+                        error={createError}
+                        setSelectAccountError={setSelectAccountError}
+                        createWorkspace={createWorkspace}
+                    />
+                </div>
+            </div>
+        </div>
+    );
+}
+
+function tryAuthorize(host: string, scopes?: string[]): Promise<SelectAccountPayload | undefined> {
+    const result = new Deferred<SelectAccountPayload | undefined>();
+    openAuthorizeWindow({
+        host,
+        scopes,
+        onSuccess: () => {
+            result.resolve();
+        },
+        onError: (error) => {
+            if (typeof error === "string") {
+                try {
+                    const payload = JSON.parse(error);
+                    if (SelectAccountPayload.is(payload)) {
+                        result.resolve(payload);
+                    }
+                } catch (error) {
+                    console.log(error);
+                }
+            }
+        },
+    }).catch((error) => {
+        console.log(error);
+    });
+    return result.promise;
+}
+
+function StatusMessage({
```

Comment:
A suggestion for a slightly different way to define a component that handles the return type automatically. You're handling a return type of `JSX.Element` here, so it won't make a big difference, but it's easy to forget the return type, and then some lint errors aren't caught.

```ts
type StatusMessageProps = {...}
const StatusMessage: FunctionComponent<StatusMessageProps> = ({ ... }) => {
```